.circleci/config.yml

```diff
@@ -0,0 +1,112 @@
+version: 2.1
+
+references:
+  images:
+    ruby: &RUBY_IMAGE circleci/ruby:2.3.4-node
+
+aliases:
+  - &restore_gem_cache
+      keys: 
+        - gemfile-{{ checksum "Gemfile.lock" }}
+  - &save_gem_cache
+      name: Save gem cache
+      key: gemfile-{{ checksum "Gemfile.lock" }}
+      paths: 
+        - ~/data/vendor/bundle
+  - &attach_workspace
+      attach_workspace:
+        at: ~/data
+
+commands:
+  bundle-install:
+    description: Bundle Install
+    parameters:
+      bundle-version:
+        type: string
+      gem-version:
+        type: string
+    steps:
+      - run: 
+          name: Upgrade bundler
+          command: |
+            gem install bundler -v '~> << parameters.bundle-version >>'
+            bundle '_<< parameters.bundle-version >>_' install
+      - run: 
+          name: Upgrade gem
+          command: |
+            sudo gem update --system '<< parameters.gem-version >>'
+      - run: 
+          name: Bundle install
+          command: |
+            bundle install --jobs 7 --retry=3 --path=vendor/bundle
+      - run: 
+          name: Bundle exec rake
+          command: |
+            bundle exec rake
+
+jobs:
+  build:
+    docker:
+      - image: *RUBY_IMAGE
+    working_directory: ~/data
+    steps:
+      - checkout
+      - *attach_workspace
+      - restore_cache: *restore_gem_cache
+      - bundle-install:
+          bundle-version: '1.17.3'
+          gem-version: '3.1.2'
+      - save_cache: *save_gem_cache
+      - persist_to_workspace:
+          root: .
+          paths:
+            - vendor/bundle
+  gem-push:
+    docker:
+      - image: *RUBY_IMAGE
+    working_directory: ~/data
+    steps:
+      - checkout
+      - *attach_workspace
+      - bundle-install:
+          bundle-version: '1.17.3'
+          gem-version: '3.1.2'
+      - run:
+          name: Setup RubyGems
+          command: bash .circleci/setup-rubygems.sh
+      - run:
+          name: Publish to RubyGems
+          command: bundle exec rake release
+  docker-push:
+    docker:
+      - image: *RUBY_IMAGE
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Upload docker image
+          command: |
+            docker login -u "${DOCKERHUB_USERNAME}" -p "${DOCKERHUB_PASSWORD}"
+            make docker
+            make docker-push
+
+workflows:
+  middleman-hashicorp:
+    jobs:
+      - build
+      - gem-push:
+          context: middleman-hashicorp
+          requires: 
+            - build
+          filters:
```

Comment:
In TeamCity the builds only happened on tags and this is [possible](https://circleci.com/docs/2.0/configuration-reference/#tags) in Circle too. Ex:

```
          filters:
            tags:
              only: /^v\d+\.\d+\.\d+$/
```

Reply:
Thanks for catching this alvin!